numba_rvsdg/core/datastructures/basic_block.py

```diff
@@ -43,34 +96,91 @@ def declare_backedge(self, target: str) -> "BasicBlock":
 
     def replace_jump_targets(self, jump_targets: Tuple) -> "BasicBlock":
         """Replaces jump targets of this block by the given tuple.
+
+        This method replaces the jump targets of  the current BasicBlock.
         The provided jump targets must be in the same order as their
         intended original replacements.
 
         Note that replacing jump targets will not replace the backedge
         tuple, so replacement for any jump targets that is declared as
-        a backedge needs to be updated separately using replace_backedges"""
+        a backedge needs to be updated separately using replace_backedges
+
+        Parameters
+        ----------
+        jump_targets: Tuple
+            The new jump target tuple. Must be ordered.
+
+        Returns
+        -------
+        result: BasicBlock
+            The resulting BasicBlock
+
+        """
         return replace(self, _jump_targets=jump_targets)
 
     def replace_backedges(self, backedges: Tuple) -> "BasicBlock":
-        """Replace back edges of this block by the given tuple.
+        """Replaces back edges of this block by the given tuple.
+
+        This method replaces the back edges of  the current BasicBlock.
         The provided back edges must be in the same order as their
-        intended original replacements."""
+        intended original replacements.
+
+        Parameters
+        ----------
+        backedges: Tuple
+            The new back edges tuple. Must be ordered.
+
+        Returns
+        -------
+        result: BasicBlock
+            The resulting BasicBlock
+
+        """
         return replace(self, backedges=backedges)
 
 
 @dataclass(frozen=True)
 class PythonBytecodeBlock(BasicBlock):
-    begin: int = None
-    """The starting bytecode offset.
+    """The PythonBytecodeBlock class is a subclass of the BasicBlock that
+    represents basic blocks with Python bytecode.
+
+    Attributes
+    ----------
+    begin: int
+        The starting bytecode offset.
+
+    end: int
+        The bytecode offset immediate after the last bytecode of the block.
     """
 
+    begin: int = None
+
     end: int = None
-    """The bytecode offset immediate after the last bytecode of the block.
-    """
 
     def get_instructions(
         self, bcmap: Dict[int, dis.Instruction]
     ) -> List[dis.Instruction]:
+        """Retrieves a list of dis.Instruction objects corresponding to
```

Comment:
```suggestion
        """Retrieves a list of `dis.Instruction` objects corresponding to
```

We can use a fixed-width font in Sphinx documenation. Though I am not sure if the correct markup is to enclose the text in single, or double backticks.